### ISSUE TITLE
[Suggestion] Links between Engineering and Science

### ISSUE THREAD

**Issue by elptito** [daid/EmptyEpsilon#260]
There are currently no link betwwen Engineering and Science station, so players don't have to discuss during the game. Because I think that Science actions could be repetitive, adding this link could be interesting.

Engineer could have one more system to take care, with an effect on the science station. I think about two possibles links : 

1- The power of the radar system could affect the scan action. bandwidth could be shorter when the power is high (scanning will be easier). And the scan could no work if the health is less than 0 %

2- The radar range could depend on the power of the radar system. 100 % of power allows a 30 U radar range, while 300 % allows a 90 U radar range (maybe less -> 100 % for 15 U, 300 % for 45 U). And the radar range could decrease with the health of the radar system.

What do you think about that ?

**Comment by Fouindor** [comment]
I think this is a good idea, I'm more biaised towards the first suggestion.

**Comment by daid** [comment]
Artemis had this link (sensor system). But I wasn't a huge fan of it. As it didn't add a meaningful mechanic. As during scanning it's not that hard to remove power from other systems to put into sensors.

**Comment by oznogon** [comment]
In Trek episodes, some of the science/engineering interactions involve integrating upgrades to ship systems. Could picking up an artifact potentially trigger science and engineering minigames to investigate and integrate an improvement? For instance, science could scan the artifact to determine what type of upgrade it is, and engineering has to move repair crew to the corresponding ship system and complete a minigame to integrate it.

Another common fiction interaction is to refuel a ship, for instance by flying through a fuel-rich gas nebula. Science could scan a nebula's composition, and engineering could calibrate and activate a scoop appropriately to boost reactor output while in the nebula.

**Comment by lunerfox** [comment]
I think Science and Engineering should have a different relationship beyond adjusting power; 

In general, Science's role should be to discover dangers and opportunities that other consoles can act on.

Science has a link with Weapons/Helm by offering information about nearby ships, which can be used to gain an advantage by adjusting frequencies and navigating out of beam range

Science has a link with Relay by receiving visual data from probes that Science can act on.

Perhaps Science should link with Engineering by means of artifacts; picking them up would allow for science to do a "brainy" puzzle to decrypt the knowledge within, gaining an upgrade point to spend for incremental improvements.

**Comment by CaptLapadite** [comment]
We played yesterday and while every one had to communicate with the engineer, science And relay officer (me) didn't have to bother with him at all. 

It would be great if the scanner was another of the energy sliders in engineering. Where boosting the signal would increase the range of the scanner. 

Another idea could be to add a tractor beam to science so he would be able to interact during fights.

**Comment by kwadroke** [comment]
With tractor beams, Engineering could set the frequency needed by Science.

**Comment by daid** [comment]
> With tractor beams, Engineering could set the frequency needed by Science.

As if engineering isn't busy enough already. I think that's the thing. Engineering never has a dull moment already.

![image](https://cloud.githubusercontent.com/assets/964186/15625746/17411c12-24b3-11e6-95c1-ba39c0bfb619.png)
I think these are the interactions that are happening. That is direct interactions that are not mission goal/tactic related, as they go trough the captain. And that involves all stations.

So saying "engineering needs X because there is no interaction between engineering and science", misses the fact that there is more inter-station interaction not happening. Which is also fine.

I'm more looking at games being played and see which station feels more involved and less involved. From that I see the following things:

* Engineering is really busy at all times. And usually fails in proper damage reports. So I'm making tweaks there to make damage more concentrated on systems instead of all over the systems.
* The shield frequency mechanic is hardly used. It takes a lot of effort to get right, and while the impact is significant, it's an invisible impact. Making my players ignore it. And I'm pondering to remove the whole mechanic...
* Relays job could be more involved. While science does not have the time to do two scans on all ships. One idea that we had was to add "sensor hacking" to relay, which can only be done if a ship is already scanned once by science, but then does (optionally) the 2nd scan by relay. Allowing you to spread that workload.

**Comment by Wrongtown** [comment]
I quite like the idea of Relay covering second scan! Some may object to it being a bit of a stretch but the concept of the science officer doing the heavy lifting and the relay officer interpreting the data works well enough :) More importantly it gives Relay something active to do and also a real reason to use probes carefully and link them to science. 

By the way I was at a session last week observing a bunch of brand new players. The physical probes hurtling off into space and also being smashed by enemy ships when they were launched right next to them was very effective :)

**Comment by CaptLapadite** [comment]
> The shield frequency mechanic is hardly used. It takes a lot of effort to get right, and while the impact is significant, it's an invisible impact. Making my players ignore it. And I'm pondering to remove the whole mechanic...

I beg to differ on that point. 
Entering in combat against an enemy of similar strength with the worse shield calibration is almost as having no shield at all. It adds a lot of pression on the decision making for the captain and it feels awesome. It forced us to leave combat fast a few times because we hadn't (or couldn't) take the time to calibrate the shields or scan properly to have the best shield frequency. I understand it may be difficult to use for a new crew, or completely ignored by an already overwhelmed new captain... but for others it is just another detail that make the game fun by the way all the team work in concert to optimize the ship for battle. Maybe simply have an option in the server to simplify the shield calibration and remove the option, or simply let it unused by the crew... they'll get to it over time. 

I also pointed a few suggestion on the forum for all stations and more importantly for the science station (radac improvement and tractor beam) . 
http://bridgesim.net/discussion/209/playtests#latest

**Comment by nallath** [comment]
The problem with the shield stuff is that it's useless against a group op enemies. They usually spawn with random shield calibrations, so there is no one "right" solution. Ignoring it is usually the best stategy. 

It could be that the system can work, it's just not working in it's current system.

**Comment by oznogon** [comment]
While poking around the beam code, I wondered if it might be a good idea to bound CpuShip beam and shield frequencies to ranges associated by group, such as by ship class/subclass or faction.

For example, all CPU fighters (or Kraylor ships) could default to shield frequency ranges from 500-600, frigates (or Exuari) from 600-700, etc. That would allow several things to happen:

-   If the Science officer identifies most or all ships in a group as being of the same class (or faction), Science can recommend one frequency good enough for all enemies in the group. This increases the value of a shield frequency change for the player ship.
-   If the science officer knows that all fighters (or Kraylor ships) have frequencies within a range, they don't need to deep-scan a ship to guess a decent shield frequency. This increases the value of an experienced Science officer without excessively penalizing an inexperienced officer. Likewise, this allows other experienced officers (namely Weapons, who controls shields and might guess an unscanned ship's class by its movement, arms, or main-screen appearance) to recognize and act on similar patterns even if Science doesn't see it. This rewards other officers whose knowledge allows them to act creatively beyond their station.
-   GMs have more tactical options when creating scenarios or adding ships. Diverse enemy squadrons become harder for players than homogenous ones, which could factor into scaling a scenario's difficulty.

**Comment by nerdcorenet** [comment]
> The problem with the shield stuff is that it's useless against a group op enemies. They usually spawn with random shield calibrations, so there is no one "right" solution. Ignoring it is usually the best stategy.
> It could be that the system can work, it's just not working in it's current system.

I disagree; When we engage a group of enemies in our games, we've found it effective to scan all enemies before engaging, and try to adjust our shields to somewhere which will average across the enemy fleet, or otherwise if there's a one-off enemy beam freq then we might engage that one first to get it out of the way. All of this adds complexity to the attack which would otherwise just be to attack each enemy in serial order.

If there is one enemy remaining for whom we are poorly calibrated we might even leave the sector and try to put some space between us for recalibration. All this adds activity to the Science station (though recently our Science officer has just been doubling on Weapons console to recalibrate the settings themself ;)

**Comment by nerdcorenet** [comment]
Also FWIW our Engineer has requested something like the original post in this Issue, such that there might be a control over Science's scanning speed/abilities as another system controlled by Engineering. This does add some complexity to Engineering and I do understand @daid's point that this station is already quite busy, but our Engineer has specifically requested we add this to enhance the immersion of the overall gameplay.

IMHO this doesn't require *interaction* between Science and Engineering directly. Depending on how you play this might just be another request the Captain makes to Engineering at appropriate times during a mission.

**Comment by oznogon** [comment]
This is much more possible than in 2016 now that sensor ranges are adjustable during the course of a scenario (they were previously unchangeable ranges). There are also enough scripting hooks now to do things similar to this as scenario-specific custom actions without changing the base game.

**Comment by srps-git** [comment]
Considering the flow, I think a better dynamic would be if Science helped Engineering - Engineering gets enough demands as is, a way for Science to help Engineering is a 'action outlet' for when Science has finished their primary things, and adds some hecticness to Science.

Consider - Science turning their scanners inwards to give Engineering data to help them fix systems better/faster, or Science can see the 'repair crew' layouts to feed to Engineering that if they arrange their repair crew in a certain pattern, their will be a buff to a system. ie; 'repair crew on front shield and beams and reactor can provide a buff that reduces the amount of heat beams generate when firing', or such.
